pkg/rosa/externalauthproviders.go

```diff
@@ -0,0 +1,136 @@
+package rosa
```

Comment:
The vendor does not support these? If it does, please use them.

Reply:
Is it not possible to provide them with a v2 client while continuing to use a v1 client for the other parts?

Reply:
The reason the two packages have separate module names is - btw - to support that:

```
$ cat main.go 
package main

import (
	"context"

	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go/aws/session"
)

func main() {
	session.NewSession()
	config.LoadDefaultConfig(context.TODO())
}

$ cat go.mod 
module github.com/openshift/hypershift/aws-test

go 1.21.5

require (
	github.com/aws/aws-sdk-go v1.51.21
	github.com/aws/aws-sdk-go-v2/config v1.27.11
)

require (
	github.com/aws/aws-sdk-go-v2 v1.26.1 // indirect
	github.com/aws/aws-sdk-go-v2/credentials v1.17.11 // indirect
	github.com/aws/aws-sdk-go-v2/feature/ec2/imds v1.16.1 // indirect
	github.com/aws/aws-sdk-go-v2/internal/configsources v1.3.5 // indirect
	github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.6.5 // indirect
	github.com/aws/aws-sdk-go-v2/internal/ini v1.8.0 // indirect
	github.com/aws/aws-sdk-go-v2/service/internal/accept-encoding v1.11.2 // indirect
	github.com/aws/aws-sdk-go-v2/service/internal/presigned-url v1.11.7 // indirect
	github.com/aws/aws-sdk-go-v2/service/sso v1.20.5 // indirect
	github.com/aws/aws-sdk-go-v2/service/ssooidc v1.23.4 // indirect
	github.com/aws/aws-sdk-go-v2/service/sts v1.28.6 // indirect
	github.com/aws/smithy-go v1.20.2 // indirect
	github.com/jmespath/go-jmespath v0.4.0 // indirect
)
```

Reply:
yeah we could, but there were other deps conflicts as well like k8s deps. short answer: It was a mess to fix that, especially that their implementation is not completely usable atm.

Reply:
The API machinery bump looks like [v0.28](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/4812465210f4d40ef41add5c87b11c1c9b1bf598/go.mod#L46) to [v0.29](https://github.com/openshift/rosa/blob/master/go.mod#L47). I wouldn't expect that to be onerous.

Reply:
its a little bit more involved than that, but as I said `UpdateExternalAuth()` func is still missing and need to be implemented as well as `PollKubeconfig()`. so, there is no point going through this hassle atm.